docs/proposals/kubelet-oci-runtime.md

```diff
@@ -0,0 +1,82 @@
+Kubelet OCI runtime integration
+===========================================================
+Authors: Mrunal Patel (@mrunalp), Vishnu Kannan (@vishh)
+
+## Abstract
+This proposal aims to introduce support for OCI Runtime Specification compatible runtimes in Kubelet.
+
+## Motivation
+Kubelet should support the OCI runtime-spec to support Open Standards. OCI Runtime Spec has the primitives to support
+creating kubernetes pods. Compatibility with OCI Specification will let kubernetes support any OCI compliant runtimes.
+
+## Design aspects
+The following subsections will discuss the various design aspects of the oci-runtime integration with kubelet. For the
+purposes of this doc, the kubelet runtime that will provide support for OCI compliant container runtimes will be
+referred to as `oci-runtime`.
+
+### Runtime API
+The implementation will make use of the kubelet Container Runtime Interface
+(https://github.com/kubernetes/kubernetes/pull/25899). The oci-runtime will be implemented against the Kubelet Container
+Runtime Interface as a standalone daemon.
```

Comment:
Yeah. Kubelet expects a server to satisfy the grpc runtime interface. Also,
OCI in its current state does not satisfy all the requirements of a
container. Kubelet requires a higher level abstraction!

On Fri, Aug 26, 2016 at 4:32 AM, Aleksa Sarai <notifications@github.com>
wrote:

> In docs/proposals/kubelet-oci-runtime.md
> <https://github.com/kubernetes/kubernetes/pull/26788#discussion_r76403980>
> :
>
> > +## Abstract
> > +This proposal aims to introduce support for OCI Runtime Specification compatible runtimes in Kubelet.
> > +
> > +## Motivation
> > +Kubelet should support the OCI runtime-spec to support Open Standards. OCI Runtime Spec has the primitives to support
> > +creating kubernetes pods. Compatibility with OCI Specification will let kubernetes support any OCI compliant runtimes.
> > +
> > +## Design aspects
> > +The following subsections will discuss the various design aspects of the oci-runtime integration with kubelet. For the
> > +purposes of this doc, the kubelet runtime that will provide support for OCI compliant container runtimes will be
> > +referred to as `oci-runtime`.
> > +
> > +### Runtime API
> > +The implementation will make use of the kubelet Container Runtime Interface
> > +(https://github.com/kubernetes/kubernetes/pull/25899). The oci-runtime will be implemented against the Kubelet Container
> > +Runtime Interface as a standalone daemon.
>
> I believe it's because you need to provide a GRPC consumer (but I wasn't
> there when @mrunalp <https://github.com/mrunalp> was trying to get this
> to work without a daemon).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26788/files/dce95c8a7564a973d5699fc7386c39dc417bfd8d#r76403980>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGvIKPZhrHkdv0uIgzq5aNoysLhO5R72ks5qjs68gaJpZM4Itvu7>
> .
>